pallets/gear-voucher/src/lib.rs

```diff
@@ -563,14 +571,57 @@ pub mod pallet {
                 !matches!(call, PrepaidCall::UploadCode { .. }),
                 Error::<T>::CodeUploadingDisabled
             );
+            ensure!(
+                !matches!(call, PrepaidCall::DeclineVoucher),
+                Error::<T>::InexistentVoucher
+            );
 
             // Looking for sponsor synthetic account.
             #[allow(deprecated)]
             let sponsor = Self::call_deprecated_sponsor(&origin, &call)
                 .ok_or(Error::<T>::UnknownDestination)?;
 
             // Dispatching call.
-            T::CallsDispatcher::dispatch(origin, sponsor, call)
+            T::CallsDispatcher::dispatch(origin, sponsor.clone(), sponsor.cast(), call)
+        }
+
+        /// Decline existing and not expired voucher.
+        ///
+        /// This extrinsic expires voucher of the caller, if it's still active,
+        /// allowing it to be revoked.
+        ///
+        /// Arguments:
+        /// * voucher_id:   voucher id to be declined.
+        #[pallet::call_index(5)]
+        #[pallet::weight(T::DbWeight::get().reads_writes(2, 4))]
+        pub fn decline(origin: OriginFor<T>, voucher_id: VoucherId) -> DispatchResultWithPostInfo {
```

Comment:
Maybe better to `cancel` or `discontinue` instead of `decline`?

Reply:
I was thinking about `reject` but it a little more rude that neutral `decline`
`discontinue` feels ultra weird
`cancel` sounds like I terminate voucher that was issued by ME. while `decline` shows that somebody "offered" me the voucher.
So I'd go with this naming atm or consider any other possibilities

Reply:
Ah, it is declined by the user, not the voucher issuer. Make sense!

Reply:
> Ah, it is declined by the user, not the voucher issuer. Make sense!

yeah, by its spender*

Reply:
The operation leads to voucher expiration, as a result it is not possible to use it, right? Maybe `Deactivate` ? `Decline` is good too.